chain/client/src/stateless_validation/state_witness_producer.rs

```diff
@@ -102,7 +103,9 @@ impl Client {
             // With stateless validation chunk producer uses recording reads when validating transactions.
             // The storage proof must be available here.
             transactions_storage_proof.ok_or_else(|| {
-                Error::Other("Missing storage proof for transactions validation".to_owned())
+                let message = "Missing storage proof for transactions validation";
+                log_assert_fail!("{message}");
```

Comment:
I'm wondering if we can make `log_assert_fail!` and `log_assert!` take arguments similar to tracing error? Can handle in independent PR.

Reply:
> take arguments similar to tracing error

this is a bit tricky since `assert` accepts narrower set or arguments, see Robin's comment: https://github.com/near/nearcore/pull/10659#discussion_r1503330188

Simonas suggested a proper solution, but it requires PhD in `tracing` library, feel free to give it a try 😅

Reply:
Let me try to invest some time into it and see if it's relatively easily crackable.

Reply:
Let me try to invest some time into it and see if it's relatively easily crackable.